src/core/core/network.cpp

```diff
@@ -280,7 +294,12 @@ namespace dsn
             _net.on_client_session_disconnected(sp);
 
             // reconnect with new socket
-            clear(++_reconnect_count_after_last_success < 3);
+            //
```

Comment:
yes - but always using false causes messages lost too often on network disconnected. it is best if we can have a better way.